src/constraint/constrSys.ml

```diff
@@ -48,6 +48,8 @@ sig
 
   val sys_change: (v -> d) -> v sys_change_info
   (** Compute incremental constraint system change from old solution. *)
+
+  val postmortem: v -> v list
```

Comment:
The description says:

> Constraint system can provide a hint in the form of the "postmortem" function, that returns for some side-effect target unknown that has become bottom, which local unknowns might now be worth re-evaluating. The purpose of this is to allow the re-evaluation of function exit nodes, when their entry becomes bottom, in order to propagate this bottom value and remove superfluous side effects along the way.

Doesn't the existing solver dependency information already have this? This essentially is a static definition of (a transitive subset of) `infl` which kind of goes against our whole dynamic dependency discovery approach.

Reply:
It is indeed some subset of the transitive subset, but it allows limiting what is re-queried rather than calling solve for all last dependencies, which is potentially very expensive.

For those with access to our drafts at https://github.com/tum-cit-pl, this is discussed in https://github.com/tum-cit-pl/combatting-precision-loss Section 4.3.

In some sense this is a hint of the constraint system to the solver in the sense of "might now be worth re-evaluating", which is a subjective thing.

We're open to changing the name but a functionality like this is needed for our paper.

Reply:
`iter_vars` seems to want to iterate through vars, but this is a much simpler query here, that just produces for the unknown corresponding to the procedure start point the endpoint. I added an explanation that applies generally to side-effecting constraint systems.

Reply:
I think we just formulated it as an `iter`-style thing at the time to avoid allocating massive lists of unknowns in the incremental setting. `iter_vars` could probably also return a `Seq.t` to achieve a similar effect.

I just had a look at `VarQuery.t` and realized that the query itself isn't polymorphic over the type of the unknown, only the returned iterator is. This probably can be generalized like `Queries.t`, but it's a bigger change than I expected, so this might have to do for now.